torch/distributions/constraints.py

```diff
@@ -458,26 +461,37 @@ def check(self, value):
         return _LowerCholesky().check(value) & unit_row_norm
 
 
-class _Symmetric(Constraint):
+class _Square(Constraint):
     """
-    Constrain to Symmetric square matrices.
+    Constrain to square matrices.
     """
     event_dim = 2
 
     def check(self, value):
+        return torch.zeros(value.shape[:-2], dtype=torch.bool).fill_(value.shape[-2] == value.shape[-1])
+
+
+class _Symmetric(_Square):
+    """
+    Constrain to Symmetric square matrices.
+    """
+
+    def check(self, value):
+        assert super().check(value).all(), \
+            "Cannot check symmetricity if the input is not a square matrix."
         return (value.transpose(-2, -1) == value).all(dim=-1).all(dim=-1)
```

Comment:
This should be `super().check(value) and...`  and the `assert` should be removed.

nit. Prefer `.mT` over `.transpose(-2, -1)` for conciseness.

Reply:
I worry about the case when `super().check(value).all()` is `False`, it may raise an error on the right part after the 'and'.
For example, if we check positive definiteness of the asymmetric matrix, we get the error eventually.
I am not sure about this part.

Reply:
Thank you for the explanation about '.mT. :) I didn't know the function before. It seems useful in matrix operations!

Reply:
True that, as there's no shortcut. That's tricky, I don't know what would be a good way of writing this, as I don't know the exact requirements of this function. cc @neerajprad

Reply:
I will try to reflect the feedback of neerajprad. Please tell me if you have some feedbacks!